flink-ai-flow/ai_flow/metric/service/metric_service.py

```diff
@@ -16,144 +16,131 @@
 
-    def getModelMetricMeta(self, request, context):
+    @catch_exception
+    def listModelMetricMetas(self, request, context):
         model_name = request.model_name
-        model_version = request.model_version
-        metric_meta = self.store.get_model_metric_meta(model_name=model_name, model_version=model_version)
-        return _warp_list_metric_meta_response(metric_meta)
+        project_name = request.project_name.value if request.HasField('project_name') else None
+        metric_metas = self.store.list_model_metric_metas(model_name=model_name, project_name=project_name)
+        return _warp_list_metric_metas_response(metric_metas)
 
-    def getMetricSummary(self, request, context):
-        metric_id = request.metric_id
-        metric_summary = self.store.get_metric_summary(metric_id=metric_id)
-        return _warp_list_metric_summary_response(metric_summary)
+    @catch_exception
+    def registerMetricSummary(self, request, context):
+        metric_summary_proto = request.metric_summary
+        metric_summary = proto_to_metric_summary(metric_summary_proto)
+        res_metric_summary = self.store.register_metric_summary(metric_summary.metric_name,
+                                                                metric_summary.metric_key,
+                                                                metric_summary.metric_value,
+                                                                metric_summary.metric_timestamp,
+                                                                metric_summary.model_version,
+                                                                metric_summary.job_execution_id)
+        return _warp_metric_summary_response(res_metric_summary)
 
-    def deleteMetricMeta(self, request, context):
-        uuid = request.uuid
-        try:
-            self.store.delete_metric_meta(uuid)
-            return Response(return_code=str(ReturnCode.SUCCESS), return_msg='', data='')
-        except Exception as e:
-            return Response(return_code=str(ReturnCode.INTERNAL_ERROR), return_msg=str(e), data='')
+    @catch_exception
+    def updateMetricSummary(self, request, context):
+        metric_summary_proto = request.metric_summary
+        metric_summary = proto_to_metric_summary(metric_summary_proto)
+        res_metric_summary = self.store.update_metric_summary(metric_summary.uuid,
+                                                              metric_summary.metric_name,
+                                                              metric_summary.metric_key,
+                                                              metric_summary.metric_value,
+                                                              metric_summary.metric_timestamp,
+                                                              metric_summary.model_version,
+                                                              metric_summary.job_execution_id)
```

Comment:
Does user know `job_execution_id` when they update `MetricSummary` ?

Reply:
`job_execution_id ` is known when updating summary for the online pipeline of ODL,